doc/connectivity/networking/api/socket_service.rst

```diff
@@ -0,0 +1,142 @@
+.. _socket_service_interface:
+
+Socket Service
+##############
+
+.. contents::
+    :local:
+    :depth: 2
+
+Overview
+********
+
+The socket service API can be used to install a handler that is called if there
+is data received to a socket. The API helps to avoid creating a dedicated thread
+for each TCP or UDP services that the application provides. Instead one thread
+is created which can serve data to multiple listening sockets which saves memory
+as only one thread needs to be created in the system in this case.
```

Comment:
```suggestion
The socket service API can be used to install a handler that is called if there
is data received on a socket. The API helps to avoid creating a dedicated thread
for each TCP or UDP service that the application provides. Instead one thread
is created that serves data to multiple listening sockets which saves memory
as only one thread needs to be created in the system in this case.
```